experimental/packages/otlp-exporter-base/src/platform/browser/OTLPExporterBrowserBase.ts

```diff
@@ -30,16 +31,21 @@ export abstract class OTLPExporterBrowserBase<
   ServiceRequest
 > extends OTLPExporterBase<OTLPExporterConfigBase, ExportItem, ServiceRequest> {
   protected _headers: Record<string, string>;
-  private _useXHR: boolean = false;
+  private sendMethod: 'beacon' | 'xhr' | 'fetch' = 'beacon';
 
   /**
    * @param config
    */
   constructor(config: OTLPExporterConfigBase = {}) {
     super(config);
-    this._useXHR =
-      !!config.headers || typeof navigator.sendBeacon !== 'function';
-    if (this._useXHR) {
+    if (!config.headers && typeof navigator.sendBeacon === 'function') {
+      this.sendMethod = 'beacon';
+    } else if (typeof XMLHttpRequest === 'function') {
+      this.sendMethod = 'xhr';
+    } else {
+      this.sendMethod = 'fetch';
+    }
```

Comment:
Deno and Cloudflare Workerd dont' support xhr inherently. Only fetch is available on those platforms. Feature detection is a good choice to support those platforms.

However, I've been thinking that we may need to add a common internal http client that can be shared across exporters -- like the zipkin exporter. Also, adding a new base http implementation currently requires dedicated work to support features like retrying https://github.com/open-telemetry/opentelemetry-js/pull/3207.

I'd propose migrating this to be an internal tool and find a minimum http interface for exporters so that we can reduce the duplication across exporters: https://github.com/open-telemetry/opentelemetry-js/pull/3577.

Reply:
Oh. I agree with creating a new shared exporter is better.
I'm OK with any method if opentelemetry-js works in a worker environment.

So... should I ask you to determine whether to drop this PR or not? 
@legendecas

Reply:
Sorry for the delayed response. I think this PR is still a good work and can be landed. https://github.com/open-telemetry/opentelemetry-js/pull/3577 updates more than OTLP exporters and needs more tailoring.